code/modules/ftl/machinery/phase_cannon.dm

```diff
@@ -176,43 +181,54 @@
 						to_chat(user, "<span class='warning'>\The [WT] must be on to complete this task!</span>")
 					return
 				playsound(src.loc, W.usesound, 50, 1)
-				to_chat(user, "<span class='notice'>You start unwelding the cannons supports...</span>")
+				to_chat(user, "<span class='notice'>You start unwelding the supports of \the [src]...</span>")
 				if(do_after(user, 20*W.toolspeed, target = src))
 					if(!src || !WT.isOn()) return
 					to_chat(user, "<span class='notice'>You unweld the supports of \the [src].</span>")
 					anchored = FALSE
 					state = CONSTRUCTION_STATE2
 					W.add_fingerprint(user)
 
-			if(istype(W, /obj/item/stack/sheet/mineral/diamond)) //Contruct
-				var/obj/item/stack/sheet/mineral/diamond/D = W
-				if(D.use(5))
-					to_chat(user, "<span class='notice'>You install \the [D] as focusing lenses for \the [src].</span>")
-					state = CONSTRUCTION_STATE4
-				else
-					to_chat(user, "<span class='warning'>You require at least 5 [D] to install lenses for \the [src]!</span>")
+			if(istype(W, /obj/item/stack/sheet)) //Contruct
+				var/obj/item/stack/sheet/S = W
+				if(istype(S, /obj/item/stack/sheet/mineral/diamond)) //|| istype(S, /obj/item/stack/sheet/glass))
+					if(S.use(5)) lens = new /obj/item/stack/sheet/mineral/diamond
+					else to_chat(user, "<span class='warning'>You require at least 5 [S] to install lenses for \the [src]!</span>")
+				else if(istype(S, /obj/item/stack/sheet/glass))
+					if(S.use(5)) lens = new /obj/item/stack/sheet/glass
+					else to_chat(user, "<span class='warning'>You require at least 5 [S] to install lenses for \the [src]!</span>")
+				else to_chat(user, "<span class='warning'>You require either 5 Glass or Diamond sheets as lenses for \the [src]!</span>")
+
+				lens.amount = 5
+				to_chat(user, "<span class='notice'>You install \the [S] as focusing lenses for \the [src].</span>")
+				state = CONSTRUCTION_STATE4
+
 
 		if(CONSTRUCTION_STATE4) //remove lenses or add chip
 			if(istype(W, /obj/item/weapon/crowbar)) //Decon
-				to_chat(user, "<span class='notice'>You pry the focusing lenses out of \the [src].</span>")
-				var/obj/item/stack/sheet/mineral/diamond/D = new (loc, 5)
-				D.add_fingerprint(user)
+				to_chat(user, "<span class='notice'>You pry \the [lens] focusing lenses out of \the [src].</span>")
+				lens.loc = src.loc
```

Comment:
im pretty sure theres a proc like put_in_hands(). i know i didnt use that for my pr either but its something to look into, isnt really necesary but might be nice.

Reply:
I'm gonna say we don't need it? Machine frames currently just dump components on the floor, so the current method just emulates that.